### ISSUE TITLE
Overrides on media queries

### ISSUE THREAD

**Issue by oyvindhermansen** [ludens-reklamebyra/react-crisscross#4]
Just a suggestion, but it would be nice to be able to override the default media queries sizes. Open for implementation suggestions ✌️

**Comment by okaune** [comment]
I guess [React Context API](https://reactjs.org/docs/context.html) is the way to go?

**Comment by okaune** [comment]
Something along the lines of:
```jsx
const mediaQueries = {
  xsmall: 0,
  small: 600,
  medium: 960,
  large: 1280,
  xlarge: 1920
};

export const LayoutContext = React.createContext(mediaQueries);

class LayoutSetting extends React.Component {
  state = {mediaQueries}
  render() {
    return (
      <LayoutContext.Provider value={this.state.mediaQueries}>
        {this.props.children}
      </LayoutContext.Provider>
    )
  }
}

class App extends React.Component {
  render() {
    return (
      <ThemeProvider>
        {/* Rest of application here */}
      </ThemeProvider>
    )
  }
}
```

**Comment by oyvindhermansen** [comment]
I totally agree on this approach. Just need to figure out the implementation details. The ThemeProvider should be able to take a settings-prop, that can handle more than media queries. By doing it that way we can scale more easily on new features with overrides!

**Comment by okaune** [comment]
So by using the approach described above and changing `mediaQueries` to `settings` will look something like:
```jsx
const settings = {
  mediaQueries: {
    xsmall: 0,
    small: 600,
    medium: 960,
    large: 1280,
    xlarge: 1920
  }
};
```

**Comment by oyvindhermansen** [comment]
Exactly!

**Comment by oyvindhermansen** [comment]
The only thing that's blocking me from implementing this, is the `LayoutContext .Consumer` which is where we get our value from the `LayoutContext.Provider` from. If we implement the feature to always come with a consumer, the Layout becomes dependent on having a Provider, and in my opinion that should be opt-in/opt-out

You got some experience in the new React Context API, @okaune ? Because I've gone through this in my head a million times now, and can't seem to figure a way passed the consumer.

**Comment by oyvindhermansen** [comment]
But how would you access the defaultValue in `createContext` without a `Consumer`? And how can you use a `Consumer` without a `Provider` ?

**Comment by oyvindhermansen** [comment]
On second thought.. I think you were right, @okaune ! Here's a proof of concept:

https://codesandbox.io/s/yw2px5k7x1

**Comment by okaune** [comment]
The reason your spreading is not working is because you are spreading the `settings` object, so the `mediaQuery` object overrides the current `mediaQuery` object. Try something like:
```jsx
{mediaQueries:{ ...settings.mediaQueries, ...props.settings.mediaQueries}
```

**Comment by oyvindhermansen** [comment]
Ah ofc! Thanks, going to give it a go real soon

**Comment by okaune** [comment]
This issue hot resolved in #24
